### DISCUSSION TITLE
Can we send huge data (1 mb) as value under single key to Redis using lettuce?

### DISCUSSION THREAD

**Discussion by Pramit2512** [redis/lettuce#2129] [Q&A]
I have a xml file of approx 1 megabyte and i want to send the content of xml as it is to redis under single key .Is there a way we can achieve this using lettuce library?

Currently when i am trying to send the data using below code  :

```
RedisFuture<?> f1= redisAsyncCommands.hset(key , key, "pasted the content of file as it is");
```
I am getting error as **"error: constant string too long"**.

Is there a way we can do this ? Or will i have to compress the file content and then send the data to redis?

**Comment by mp911de** [comment]
Can you provide the full stack trace?

**Comment by Pramit2512** [comment]
Hi Mark,

My application is failing in build . So  i am getting below build error:
```
error: constant string too long  redisAsyncCommands.set(mykey,"<ProductOffering>\n" +.....
```

Below is my code :
               ` redisAsyncCommands.set(mykey,"my json data" +" .........."+".............."+".......this continues with 29,000 lines of data......"+"..");`

**Comment by jreilly650** [comment]
I think you are hitting java limitations. Back many years ago I remember
seeing generated java files hitting a limit of 64k file size, but based on
the error message it sounds like there is a limit on string constant size.
You could just put the string content into a resource file and load from
the classpath instead.
